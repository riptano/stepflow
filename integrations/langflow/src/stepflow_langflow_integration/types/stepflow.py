"""Stepflow workflow and step type definitions."""

from dataclasses import dataclass, field
from typing import Dict, Any, List, Optional


@dataclass
class StepflowStep:
    """Represents a Stepflow workflow step."""
    
    id: str
    component: str
    input: Dict[str, Any]
    output_schema: Optional[Dict[str, Any]] = None
    
    def to_dict(self) -> Dict[str, Any]:
        """Convert step to dictionary representation."""
        result = {
            "id": self.id,
            "component": self.component,
            "input": self.input,
        }
        
        if self.output_schema:
            result["output_schema"] = self.output_schema
        
        return result


@dataclass 
class StepflowWorkflow:
    """Represents a Stepflow workflow."""
    
    name: str
    steps: List[StepflowStep]
    description: Optional[str] = None
    version: str = "1.0"
    schema: str = "https://stepflow.org/schemas/v1/flow.json"
    output: Optional[Dict[str, Any]] = None
    
    def to_dict(self) -> Dict[str, Any]:
        """Convert workflow to dictionary representation."""
        result = {
            "schema": self.schema,
            "name": self.name,
            "steps": [step.to_dict() for step in self.steps],
        }
        
        if self.description:
            result["description"] = self.description
        
        if self.version != "1.0":
            result["version"] = self.version
        
        if self.output:
            result["output"] = self.output
        
        return result