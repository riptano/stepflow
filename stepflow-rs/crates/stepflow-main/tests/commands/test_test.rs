use super::stepflow;
use insta_cmd::assert_cmd_snapshot;

#[test]
fn test_test_help() {
    assert_cmd_snapshot!(stepflow().arg("test").arg("--help"));
}

#[test]
fn test_test_basic_workflow() {
    assert_cmd_snapshot!(stepflow().arg("test").arg("tests/basic.yaml"));
}

#[test]
fn test_test_mock_directory() {
    assert_cmd_snapshot!(stepflow().arg("test").arg("tests/mock"));
}

#[test]
fn test_test_with_custom_config() {
    assert_cmd_snapshot!(
        stepflow()
            .arg("test")
            .arg("--config=tests/builtins/stepflow-config.yml")
            .arg("tests/builtins")
    );
}

#[test]
fn test_test_nonexistent_path() {
    assert_cmd_snapshot!(stepflow().arg("test").arg("nonexistent_path"));
}
