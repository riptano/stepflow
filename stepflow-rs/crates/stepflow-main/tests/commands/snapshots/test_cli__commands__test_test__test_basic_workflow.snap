---
source: crates/stepflow-main/tests/commands/test_test.rs
info:
  program: stepflow
  args:
    - "--log-file=/dev/null"
    - "--omit-stack-trace"
    - test
    - tests/basic.yaml
---
success: false
exit_code: 1
----- stdout -----

----- stderr -----
Missing file: tests/basic.yaml
